docs/orchestrating-elastic-stack-applications/enterprise-search.asciidoc

```diff
@@ -0,0 +1,242 @@
+spec:
+  version: {version}
+  count: 1
+  elasticsearchRef:
+    name: quickstart
+  config:
+    # define the exposed URL at which users will reach Enterprise Search
+    ent_search.external_url: https://my-custom-domain:3002
+    # configure app search document size
+    app_search.engine.document_size.limit: 100kb
+----
+
+[id="{p}-enterprise-search-secret-configuration"]
+=== Reference Kubernetes secrets for sensitive settings
+
+Some sensitive settings are best stored in Kubernetes secrets, referenced in the Enterprise Search specification.
+
+This example sets up a Secret with SMTP credentials:
+
+[source,yaml,subs="attributes"]
+----
+apiVersion: enterprisesearch.k8s.elastic.co/v1beta1
+kind: EnterpriseSearch
+metadata:
+  name: enterprise-search-quickstart
+spec:
+  version: {version}
+  count: 1
+  elasticsearchRef:
+    name: quickstart
+  config:
+    ent_search.external_url: https://my-custom-domain:3002
+  configRef:
+  - secretName: smtp-credentials
+---
+kind: Secret
+apiVersion: v1
+metadata:
+  name: smtp-credentials
+stringData:
+  enterprise-search.yml: |-
+    email.account.enabled: true
+    email.account.smtp.auth: plain
+    email.account.smtp.starttls.enable: false
+    email.account.smtp.host: 127.0.0.1
+    email.account.smtp.port: 25
+    email.account.smtp.user: myuser
+    email.account.smtp.password: mypassword
+    email.account.email_defaults.from: my@email.com
+----
+
+ECK merges the content of `config` and `configRef` into a single internal Secret. In case of duplicate settings, the last referenced Secret takes precedence.
+
+[id="{p}-enterprise-search-custom-pod-template"]
+=== Customize the Pod template
+
+Enterprise Search Pods specification can be overridden through the `podTemplate` element.
+
+This example overrides the default 4Gi deployment to use 8Gi instead, and makes the deployment highly-available with 3 Pods:
+
+[source,yaml,subs="attributes"]
+----
+apiVersion: enterprisesearch.k8s.elastic.co/v1beta1
+kind: EnterpriseSearch
+metadata:
+  name: enterprise-search-quickstart
+spec:
+  version: {version}
+  count: 3
+  elasticsearchRef:
+    name: quickstart
+  podTemplate:
+    spec:
+      containers:
+      - name: enterprise-search
+        resources:
+          requests:
+            cpu: 3
+            memory: 8Gi
+          limits:
+            memory: 8Gi
+        env:
+        - name: JAVA_OPTS
+          value: -Xms7500m -Xmx7500m
+----
+
+[id="{p}-enterprise-search-expose"]
+=== Expose Enterprise Search
+
+By default ECK manages self-signed TLS certificates to secure the connection to Enterprise Search.
+It also restricts the Kubernetes service to `ClusterIP` type that cannot be accessed publicly.
+
+See link:../k8s-accessing-elastic-services.html[how to access Elastic Stack services] to customize TLS settings and expose the service.
+
+NOTE: When exposed outside the scope of `localhost`, make sure to set `ent_search.external_url` accordingly in Enterprise Search configuration.
```

Comment:
```suggestion
NOTE: When exposed outside the scope of `localhost`, make sure to set `ent_search.external_url` accordingly in the Enterprise Search configuration.
```

Reply:
I wonder if we should make this clearer that really, any usable configuration needs this

Reply:
do you think this note is not explicit enough?

Reply:
I think it could be clearer, the [docs](https://www.elastic.co/guide/en/enterprise-search/current/configuration.html) contain some decent phrasing imo:

>Define the exposed URL at which users will reach Enterprise Search.
Defaults to localhost:3002 for testing purposes.
Most cases will use one of:
* An IP: http://255.255.255.255
* A FQDN: http://example.com
* Shortname defined via /etc/hosts: http://ent-search.search
ent_search.external_url: http://localhost:3002